src/arch/amdgpu/vega/gpu_mem_helpers.hh

```diff
@@ -129,6 +129,55 @@ initMemReqHelper(GPUDynInstPtr gpuDynInst, MemCmd mem_req_type,
     }
 }
 
+template<int N>
+inline void
+initScratchReqHelper(GPUDynInstPtr gpuDynInst, MemCmd mem_req_type)
+{
+    int req_size = N * sizeof(VegaISA::VecElemU32);
+    int block_size = gpuDynInst->computeUnit()->cacheLineSize();
+
+    gpuDynInst->resetEntireStatusVector();
+    for (int lane = 0; lane < VegaISA::NumVecElemPerVecReg; ++lane) {
+        if (gpuDynInst->exec_mask[lane]) {
+            Addr vaddr[N];
```

Comment:
Do we need to check in here if a lane is active?

Reply:
Yes, It is checking exec mask here. For swizzled accesses there will be multiple `vaddr`s per-lane so it looks a bit different than global accesses

Reply:
I must have missed it, thanks.